lib/rodauth/features/jwt.rb

```diff
@@ -83,7 +93,55 @@ def set_notice_now_flash(message)
     end
 
     def session_jwt
-      JWT.encode(session, jwt_secret, jwt_algorithm)
+      JWT.encode(jwt_claims.merge(:data=>session), jwt_secret, jwt_algorithm)
+    end
+
+    def jwt_claims
+      claims = {}
+      %i[aud exp iat iss jti nbf sub].each do |reserved_claim|
+        next unless claim = send("jwt_#{reserved_claim}")
+
+        claims[reserved_claim] =
+          if claim.respond_to?(:call)
+            claim.call(self)
```

Comment:
Is passing the Rodauth instance to the proc the best way for the caller to introspect the application to generate a jti/sub/aud/etc. claim? The one big caveat is that you can't access the session when verifying the session (it recurses infinitely), so we either need a guard against that or some kind of warning in the documentation.